elip-21mil.mediawiki

```diff
@@ -0,0 +1,99 @@
+
+==Design==
+
+===Overview===
+
+To enable the unblinded issuance and reissuance of assets other than the <code>pegged_asset</code> with values greater than <code>MAX_MONEY</code> (21 million) the <code>MoneyRange</code> check for explicit values in the <code>VerifyIssuanceAmount</code> consensus rule will be modified to only apply when the <code>asset</code> is <code>pegged_asset</code>. 
+
+To enable the inclusion of this consensus rule change in Elements releases without risking a network fork, a new policy rule preventing the unblinded issuance and reissuance of assets (and reissuance tokens) with values greater than <code>MAX_MONEY</code> before the the technical consensus hard fork is activated. This policy rule will be configurable but applied as default in all releases until the hard fork activation. 
+
+===Drawbacks===
+
+Enabling this feature requires a hard-fork, which has to be carefully coordinated and has the potential to cause disruption. To mitigate this disruption, new releases of Elements before the hard-fork activation can include the consensus changes to <code>VerifyIssuanceAmount</code> but any issuance/reissuance transactions that would trigger a network fork will be prevented from being included in a block with the new default policy rule. 
+
+===Specification===
+
+In the function <code>VerifyIssuanceAmount</code> where <code>value</code> is explicit, the value is only checked for <code>MoneyRange(value)</code> when the <code>asset</code> is equal to the consensus parameter <code>pegged_asset</code>. 
+
+A new function <code>IsIssuanceInMoneyRange(tx)</code> returns true if a transaction has explicit issuance and reissuance values (and reissuance token values) are within <code>MoneyRange(value)</code>, otherwise false. This function is evaluated for transactions in <code>MemPoolAccept</code> if <code>fAcceptUnlimitedIssuances</code> is false. If this function returns false, the transaction is rejected with error message: <code>"issuance-out-of-range""</code>. 
+
+<code>fAcceptUnlimitedIssuances</code> is set to false by default, but can be set to true with the new config option <code>-acceptunlimitedissuances=1</code>
+
+==Backwards Compatibility==
+
+The change to <code>VerifyIssuanceAmount</code> enabling unblinded issuances greater than 21 million is a consensus hard fork, however the <code>fAcceptUnlimitedIssuances</code> with default false will prevent any transaction that would cause a block to be invalid to older, unupgraded nodes. When a node is configured with <code>-acceptunlimitedissuances=1</code> it can potentially create blocks (by including issuance transactions) that would be invalid to unupgraded nodes. 
+
+==Reference Implementation==
+
+https://github.com/ElementsProject/elements/pull/1445
+https://github.com/ElementsProject/elements/pull/1507
+
+==Test Vectors==
```

Comment:
In f35d8c66c363277ae2d444d845e1db0a39d687b5:

I get the motiation for these test vectors but in practice all nodes will reject all of these due to missing inputs. (And if you construct transactions that *would* be accepted then some joker will broadcast them and then they'll stop working.)

Maybe we still want to retain them, but we should provide instructions for using them.